cpskin/menu/browser/menu.py

```diff
@@ -285,6 +285,8 @@ def _submenu(self, items, tabindex, menu_level=0, menu_classnames='', close_butt
         return submenu
 
     def _menuitem(self, item, tabindex, first=False, last=False, menu_level=0):
+        portal_membership = api.portal.get_tool('portal_membership')
+        can_edit = portal_membership.checkPermission('Modify portal content', self.context)
```

Comment:
I think, "can edit" should be check on "menu item" object and not on context, it should be tested after these lines 

    if brain.Type == 'Link':
            obj = brain.getObject()

We would like to test if a user can edit menu and not context